### DISCUSSION TITLE
p:messages: change markup for a better accessibility

### DISCUSSION THREAD

**Discussion by hgfrey** [primefaces/community#898] [PrimeFaces] [answered]
In our accessibility test, the markup rendered for p:messages was criticized. The testers expect a button instead of a link. One possibility would be to set the aria-role attribute with the value button via passthrough. But this is then rendered in the wrong place, namely two levels higher in the surrounding div.

```
 <div id="messages" class="ui-messages ui-widget ui-g-12 ui-md-12 ui-lg-12 ui-no-margin" aria-live="polite" aria-role="button">
   <div class="ui-messages-error ui-corner-all">
     <a href="#" class="ui-messages-close" onclick="$(this).parent().slideUp();return false;" aria-label="Meldungen Schließen">
       <span class="ui-icon ui-icon-close"></span>
    </a>
    <span class="ui-messages-error-icon"></span>
    <ul>
      <li role="alert" aria-atomic="true">
        <span class="ui-messages-error-summary">Bitte geben Sie einen Infotext an. </span>
      </li>
    </ul> 
  </div>
 </div>

```
Is there any idea how I can meet the accessibility test here?

**Comment by melloware** [comment]
I am pretty sure the link probably just needs `role="button"` for the close button.  Please open a GitHub ticket.

AS a note i think we have this pattern in a few places like Sidebar close button so I would want to fix them all.

**Reply by melloware** [reply, marked as answer]
Fixed with PR: https://github.com/primefaces/primefaces/pull/11276

For you can simply use jQuery to do it...

```js
$('.ui-messages-close').attr('role', 'button');
```
